prow/plugins/updateconfig/updateconfig.go

```diff
@@ -119,7 +117,19 @@ func Update(fg FileGetter, kc corev1.ConfigMapInterface, name, namespace string,
 			return fmt.Errorf("get file err: %v", err)
 		}
 		logger.WithFields(logrus.Fields{"key": key, "filename": filename}).Debug("Populating key.")
-		cm.Data[key] = string(content)
+		value := string(content)
+		if gzipContent {
+			buff := bytes.NewBuffer([]byte{})
+			// TODO(bentheelder): this error is wildly unlikely for anything that
+			// would actually fit in a configmap, we could just as well return
+			// the error instead of falling back to the raw content
+			if _, err := gzip.NewWriter(buff).Write(content); err != nil {
+				logger.WithError(err).Error("failed to gzip content, falling back to raw")
+			} else {
+				value = buff.String()
+			}
+		}
+		cm.Data[key] = value
```

Comment:
does it make sense to log the value of `value` here?

Reply:
We didn't before, and logging MBs of yaml or lots of gzipped data sounds terrible.

Reply:
no, no. i meant to log the size of `value` when writing it to the config map.
not logging `value` itself.

Reply:
Sorry I missed your edit or something 😅
Sure? OTOH that's pretty easy to go inspect 🤷‍♂️ tried to make minimally invasive changes.